tools/print-change-list.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+from subprocess import run
+import textwrap
+from os import getenv
+from subprocess import PIPE
+from sys import exit, stderr
+
+import dateutil.parser
+import requests
+
+GITHUB_API_URL = 'https://api.github.com'
+
+
+class PRInfo(object):
+    def __init__(self, id, title):
+        self.id = id
+        self.title = title
+
+
+def pr_link(repo, id):
+    """Return a link to a GitHub Pull Request in markdown format"""
+
+    return '[#{id}](https://github.com/{repo}/pull/{id})'.format(id=id, repo=repo)
+
+
+def get_last_tag():
+    proc = run(['git','tag','--list','--sort=-taggerdate'],
+        stdout=PIPE)
+    tags = proc.stdout.decode().splitlines()
+
+    if tags:
+        return tags[0]
+    else:
+        return None
+
+
+def get_tag_date(tag):
+    tag_date_result = run(['git', 'tag', '--list', tag, '--format=%(taggerdate)'],
+        stdout=PIPE)
+    tag_date_str = tag_date_result.stdout.decode().strip()
+    return dateutil.parser.parse(tag_date_str)
+
+
+def github_request(auth_token, repo, path, **kwargs):
```

Comment:
Most of the time this script works fine without a token. However if there have been a large number of changes since the last tag, it is possible that the script may hit the rate limit for non-authenticated requests. Supporting an auth token also enables it to work for private repos.